test-e2e/reverse-proxy.js

```diff
@@ -1,33 +1,70 @@
-/**
- * This is a simple reverse proxy that makes sure that any request to localhost:3333 is forwarded to localhost:3000.
- *
- * This mimicks the type of setup you need in nginx or fronting server to the helia-service-worker-gateway in order
- * to handle subdomain requests and origin isolation.
- */
-import httpProxy from 'http-proxy'
+/* eslint-disable no-console */
```

Comment:
new native reverse-proxy using node:http..

Pretty simple, but I would have rather used a proxy library if one existed.. most are middleware or other more complex things that we don't need.